tensorflow_addons/optimizers/tests/moving_average_test.py

```diff
@@ -169,3 +176,104 @@ def test_fit_simple_linear_model():
 
     max_abs_diff = np.max(np.abs(predicted - y))
     assert max_abs_diff < 5e-3
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("sequential_update", [True, False])
+def test_start_step(sequential_update):
+    for sequential_update in [True, False]:
+        var0 = tf.Variable([1.0, 2.0])
+        grads0 = tf.constant([0.1, 0.1])
+        grads_and_vars = [(grads0, var0)]
+
+        opt = MovingAverage(
+            tf.keras.optimizers.SGD(lr=1.0),
+            sequential_update=sequential_update,
+            average_decay=0.5,
+            start_step=1,
+        )
+
+        opt.apply_gradients(grads_and_vars)
+
+        np.testing.assert_allclose(var0.read_value(), [0.9, 1.9])
+
+        ema_var0 = opt.get_slot(var0, "average")
+        if sequential_update:
+            np.testing.assert_allclose(ema_var0.read_value(), [0.9, 1.9])
+
+        opt.apply_gradients(grads_and_vars)
+
+        np.testing.assert_allclose(var0.read_value(), [0.8, 1.8])
+
+        if sequential_update:
+            np.testing.assert_allclose(ema_var0.read_value(), [0.85, 1.85])
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("sequential_update", [True, False])
+def test_dynamic_decay(sequential_update):
+    for sequential_update in [True, False]:
+        var0 = tf.Variable([1.0, 2.0])
+        grads0 = tf.constant([0.1, 0.1])
+        grads_and_vars = [(grads0, var0)]
+
+        opt = MovingAverage(
+            tf.keras.optimizers.SGD(lr=2.0),
+            sequential_update=sequential_update,
+            average_decay=0.5,
+            dynamic_decay=True,
+        )
+
+        opt.apply_gradients(grads_and_vars)
+        opt.apply_gradients(grads_and_vars)
+
+        np.testing.assert_allclose(var0.read_value(), [0.6, 1.6])
+
+        ema_var0 = opt.get_slot(var0, "average")
+        if sequential_update:
+            np.testing.assert_allclose(ema_var0.read_value(), [0.64, 1.64])
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("sequential_update", [True, False])
+def test_swap_weights(sequential_update):
+    for sequential_update in [True, False]:
+
+        strategy = tf.distribute.MirroredStrategy()
```

Comment:
We can't use distributed strategies in the test suite yet. It's work in progress, a first step is there: #1713

Reply:
Can I use OneDeviceStrategy here?

Reply:
We're going to have very soon distributed capabilities in the tests, could you take a look at #1770 and tell me if this would work for you?

Reply:
That should work for this test case. Do you know when those patches will be merged?

Reply:
Two pull requests need to get reviewed to get there. So I'd expect a week or so. If next week it's not merged, I'll ping the other maintainers about it.

Reply:
Since the blocking pull requests are still under review, would it be possible to merge this in with either the test commented out or just using OneDeviceStrategy in the test? I could immediately make a pull request that fixed the test case so it could be merged once the blocking pull requests are submitted.

Reply:
Would either of these options work for you? We would like to get these changes in soon because they are fixing real problems with MovingAverage.